### PR TITLE
Fix HTTP statuses

### PR DESCRIPTION
**Pull request by stereobooster**
Fix for https://github.com/peterbe/minimalcss/issues/162

WDYT?

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -9,6 +9,15 @@ const cheerio = require('cheerio');
 const utils = require('./utils');
 const url = require('url');
 
+const isOk = status => status === 200 || status === 304;
```
Comment (peterbe) [inline src/run.js]: We should probably stick closer to how [puppeteer does it](https://github.com/GoogleChrome/puppeteer/blob/38c6873fbbc469fe422977131bf2da8e75ec9228/lib/NetworkManager.js#L554)

Perhaps, instead of our own `isOk` function, we do something like:
```javascript
if (!(response.ok() || response.status() === 304)) {
```

Kinda strange the puppeteer doesn't consider 304 to be "OK". 

Also, we could do something like 
```javascript
if (response.status() >= 300 && response.headers().location) {
   ... it wants to redirect...
```

**Comment by stereobooster** [comment]
> It hurts my soul to see such a specific list of specific error codes

but they are specified by standard

```
300 Multiple Choices
301 Moved Permanently
302 Found
303 See Other
304 Not Modified
305 Use Proxy
307 Temporary Redirect
308 Permanent Redirect
```

I picked those that make sense

File: src/run.js (hunk 1 above)
Comment (stereobooster) [reply src/run.js]: note: this is how ok function looks like

```js
  ok() {
    return this._status === 0 || (this._status >= 200 && this._status <= 299);
  }
```

**Comment by peterbe** [comment]
Sorry, I should have put [this comment](https://github.com/peterbe/minimalcss/issues/162#issuecomment-374370297) here in the PR instead. My bad. But still, what do you think about that idea?

**Comment by stereobooster** [comment]
updated code

File: src/run.js [hunk 2]
```diff
@@ -185,7 +187,7 @@ const processPage = ({
           return safeReject(
             new Error(`${response.status()} on ${responseUrl}`)
           );
-        } else if (response.status() >= 300) {
+        } else if (response.status() >= 300 && response.status() !== 304) {
```
Comment (peterbe) [inline src/run.js]: Can you move this second if statement part *inside*. So it's like this:
```javascript
} else if (response.status() >= 300) {
  // 304 Not Modified is an exception to the 3xx responses.
  // It's expected to not be a redirect and not have content. 
  if (response.status() !== 304) {
    ...existing code...
  }
}
```
Otherwise, you'll end up in the `} else if (resourceType === 'stylesheet') {` around line 199. 

Look at this screenshot from Firefox's Dev tools. 
screen shot 2018-03-20 at 7 34 49 am
It's smart enough to figure out that the response type is "stylesheet" but if that was to happen in `minimalcss` it would fail unpredictably on the line `response.text().then(text => {`.
Basically for 304 Not Modified we don't want to write down their redirects *or* try to read their content.

Comment (stereobooster) [reply src/run.js]: But for 304 we should read content. Server didn't send content - yes, but browser should have this content, so `response.text()` should work. Unless I'm confusing something - we should treat 304 as 200

Comment (peterbe) [reply src/run.js]: A HA! That makes sense. You're right. Arguably we should have the content already if the request was made such that the server responded with a 304. But perhaps if you pass in a `puppeteer` browser instance that already has some stuff cached, it could happen. 
I rest my case.
